src/commands/kick.ts

```diff
@@ -177,10 +186,19 @@ async function kickHandler(interaction: ChatInputCommandInteraction): Promise<vo
 		}).catch(() => console.log('Failed to DM user'));
 
 		if (isKick) {
-			await member.kick(reason);
+			if (deleteMessages) {
+				await member.ban({
+					reason: `Soft-ban: ${reason}`,
+					deleteMessageSeconds: deleteMessages ?? undefined
+				});
+				await guild.members.unban(member, 'Soft-ban removed');
+			} else {
+				await member.kick(reason);
+			}
```

Comment:
Maybe I'm missing something but what exactly is the point of calling this a "soft ban"? this doesn't really make any sense to me, it's just a kick why call it anything else?

Reply:
The mod who suggested it called it a "soft ban," and I think that "soft ban" is established terminology in other Discord moderation bots for a kick with a message purge.

Reply:
I've never heard of this before, and when Googling for it I found several places all using the term "soft ban" to mean very different things, including:

- Kicks with message deletions (as used here)
- Timeouts (both in terms of limited visibility and messages)
- Hiding user sent messages from other users (not deleting the, akin to a "shadow ban" on platforms like Twitter)

This feels like confusing terminology to me since there's wildly different meanings for the phrase depending on where you look. I'm in favor of just calling it what it is, a kick with message deletions

Reply:
That sounds fair, will change.